File: frontend/src/types/rsyslog-stats.ts
export interface RsyslogStatsComponent {
  collected_at: string
  origin: string
  name: string
  stats: Record<string, number>
}

export interface RsyslogStatsSummary {
  total_submitted: number
  total_processed: number
  total_failed: number
  total_suspended: number
  main_queue_size: number
  main_queue_max_size: number
  total_discarded: number
  filter_rate: number
  failure_rate: number
  ingest_rate: number
  components: RsyslogStatsComponent[]
}

export interface RsyslogStatsSummaryResponse {
  data: RsyslogStatsSummary
}

export interface RsyslogStatsTimeSeries {
  time: string
  name: string
  value: number
}

export interface RsyslogStatsVolumeResponse {
  data: RsyslogStatsTimeSeries[]
}

/** Flat record for Unovis charts: x (epoch ms) and dynamic name keys. */
export interface RsyslogStatsDataRecord {
  x: number
  [name: string]: number
}

File: api/internal/model/rsyslog_stats.go
package model

import (
	"encoding/json"
	"time"
)

// RsyslogStatsSummary contains aggregated KPIs for a time range.
type RsyslogStatsSummary struct {
	TotalSubmitted   int64                   `json:"total_submitted"`
	TotalProcessed   int64                   `json:"total_processed"`
	TotalFailed      int64                   `json:"total_failed"`
	TotalSuspended   int64                   `json:"total_suspended"`
	MainQueueSize    int64                   `json:"main_queue_size"`
	MainQueueMaxSize int64                   `json:"main_queue_max_size"`
	TotalDiscarded   int64                   `json:"total_discarded"`
	FilterRate       float64                 `json:"filter_rate"`
	FailureRate      float64                 `json:"failure_rate"`
	IngestRate       float64                 `json:"ingest_rate"`
	Components       []RsyslogStatsComponent `json:"components"`
}

// RsyslogStatsComponent is the latest snapshot for one rsyslog component.
type RsyslogStatsComponent struct {
	CollectedAt time.Time       `json:"collected_at"`
	Origin      string          `json:"origin"`
	Name        string          `json:"name"`
	Stats       json.RawMessage `json:"stats"`
}

// RsyslogStatsTimeSeries is one time-bucketed data point.
type RsyslogStatsTimeSeries struct {
	Time  time.Time `json:"time"`
	Name  string    `json:"name"`
	Value float64   `json:"value"`
}
